masonry/src/widgets/indexed_stack.rs

```diff
@@ -0,0 +1,382 @@
+        BorderColor::prop_changed(ctx, property_type);
+        BorderWidth::prop_changed(ctx, property_type);
+        CornerRadius::prop_changed(ctx, property_type);
+        Padding::prop_changed(ctx, property_type);
+    }
+
+    fn layout(
+        &mut self,
+        ctx: &mut masonry_core::core::LayoutCtx<'_>,
+        props: &mut masonry_core::core::PropertiesMut<'_>,
+        bc: &masonry_core::core::BoxConstraints,
+    ) -> vello::kurbo::Size {
+        let border = props.get::<BorderWidth>();
+        let padding = props.get::<Padding>();
+
+        let bc = *bc;
+        let bc = border.layout_down(bc);
+        let bc = padding.layout_down(bc);
+
+        let origin = Point::ORIGIN;
+        let origin = border.place_down(origin);
+        let origin = padding.place_down(origin);
+
+        if !self.children.is_empty() && self.active_child >= self.children.len() {
+            debug_panic!(
+                "IndexedStack active child index ({}) is not within the children vector (len {})",
+                self.active_child,
+                self.children.len()
+            );
+        }
+        let mut child_size = bc.min();
+        for (idx, child) in self.children.iter_mut().enumerate() {
+            if idx == self.active_child {
+                ctx.set_stashed(child, false);
+                let child_bc = bc;
+                child_size = ctx.run_layout(child, &child_bc);
+                ctx.place_child(child, origin);
+            } else {
+                ctx.set_stashed(child, true);
+                ctx.skip_layout(child);
+            }
+        }
+
+        child_size
+    }
+
+    fn paint(
+        &mut self,
+        ctx: &mut masonry_core::core::PaintCtx<'_>,
+        props: &masonry_core::core::PropertiesRef<'_>,
+        scene: &mut vello::Scene,
+    ) {
+        let border_width = props.get::<BorderWidth>();
+        let border_radius = props.get::<CornerRadius>();
+        let bg = props.get::<Background>();
+        let border_color = props.get::<BorderColor>();
+
+        let bg_rect = border_width.bg_rect(ctx.size(), border_radius);
+        let border_rect = border_width.border_rect(ctx.size(), border_radius);
+
+        let brush = bg.get_peniko_brush_for_rect(bg_rect.rect());
+        fill(scene, &bg_rect, &brush);
+        stroke(scene, &border_rect, border_color.color, border_width.width);
+
+        // paint the baseline if we're debugging layout
+        if ctx.debug_paint_enabled() && ctx.baseline_offset() != 0.0 {
+            let color = ctx.debug_color();
+            let my_baseline = ctx.size().height - ctx.baseline_offset();
+            let line = Line::new((0.0, my_baseline), (ctx.size().width, my_baseline));
+
+            let stroke_style = Stroke::new(1.0).with_dashes(0., [4.0, 4.0]);
+            scene.stroke(&stroke_style, Affine::IDENTITY, color, None, &line);
+        }
+    }
+
+    fn accessibility_role(&self) -> Role {
+        Role::GenericContainer
```

Comment:
There are reasonable arguments for TabPanel here, but ultimately this is going to need more passes to get right, so using GenericContainer here now is reasonable